reverse_engineering/connection_settings_modal/connectionSettingsModalConfig.json

```diff
@@ -8,14 +8,46 @@
 				"inputType": "text",
 				"inputPlaceholder": "Name"
 			},
+			{
+				"inputLabel": "Mode",
+				"inputKeyword": "mode",
+				"description": "Specify connection mode",
+				"inputType": "select",
+				"defaultValue": "thin",
+				"options": [
+					{
+						"value": "thin",
+						"label": "Thin mode"
+					},
+					{
+						"value": "thick",
+						"label": "Thick mode",
+						"dependency": {
+							"dependantOnOS": true,
+							"platform": "macOS",
+							"architecture": "arm"
+						}
+					}
+				]
+			},
 			{
 				"inputLabel": "Client type",
 				"inputKeyword": "clientType",
 				"inputType": "select",
 				"defaultValue": "Host / Port",
 				"options": [
-					{ "value": "OracleHome", "label": "Oracle Home" },
-					{ "value": "InstantClient", "label": "Instant Client" }
+					{
+						"value": "OracleHome",
+						"label": "Oracle Home"
+					},
+					{
+						"value": "InstantClient",
+						"label": "Instant Client",
+						"dependency": {
```

Comment:
"Client type" and "Client location" inputs should be hidden and not the select option.